test/integration/controller/core/localqueue_controller_test.go

```diff
@@ -66,11 +66,11 @@ var _ = ginkgo.Describe("Queue controller", func() {
 	})
 
 	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
 		for _, rf := range resourceFlavors {
-			gomega.Expect(util.DeleteResourceFlavor(ctx, k8sClient, &rf)).To(gomega.Succeed())
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, &rf, true)
```

Comment:
Question to check my understanding, is this because the ClusterQueue and RF use finalizers to clean up before actual deletion, so that, the actual deletion is asynchronous and we need to await until the Get for the objects returns "Not found".